### ISSUE TITLE
Console tab doesn't navigate to current repository directory

### ISSUE THREAD

**Issue by ndrscrls** [gitextensions/gitextensions#5861]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
When I open a repository, or when I switch to a different one in the UI, the console stays at the ` ~` directory. 
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Choose a repository in the home screen, or switch to a different one in the repository dropdown.
**What is the expected behavior?**
A `cd` command is issued to the Console so it will switch automatically to the current directory.
**Environment you encounter the issue:**
Console Tab.

- Git Extensions 3.00.00.4433
- Build fca7cf228b481ee8c1b779cf7b882ccdfbdcd1bc
- Git 2.19.1.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.2116.0
- DPI 96dpi (no scaling)

**Did this work in previous version of GitExtensions (which)?**
2.5

**Comment by vbjay** [comment]
Works for me.  
- Git Extensions 3.00.00.4433
- Build fca7cf228b481ee8c1b779cf7b882ccdfbdcd1bc
- Git 2.19.1.windows.1
- Microsoft Windows NT 6.3.9600.0
- .NET Framework 4.7.3221.0
- DPI 96dpi (no scaling)

![image](https://user-images.githubusercontent.com/3629489/49660608-016d5600-fa15-11e8-9156-cab338e929e6.png)

**Comment by vbjay** [comment]
We also have an issue template for a reason.

**Comment by ndrscrls** [comment]
I updated the issue to follow the template, I'm sorry about that. I think the problem is the `cd` command is not being issued, this is all I see when switching repos, no `cd` command: 
![image](https://user-images.githubusercontent.com/10979446/49661527-45f9f100-fa17-11e8-8bf1-e37e0263ced8.png)

**Comment by vbjay** [comment]
How are you switching?  Do this.

-Close GE
- Open GE
- Open a repo
- switch to console tab
- Use dropdown to switch to another repo and wait.  
![image](https://user-images.githubusercontent.com/3629489/49661858-3fb84480-fa18-11e8-95af-64cbe27feea2.png)
- Give it a sec to load and issue cd command

**Comment by vbjay** [comment]
![image](https://user-images.githubusercontent.com/3629489/49662120-03d1af00-fa19-11e8-8386-0d56edd43d4b.png)
![image](https://user-images.githubusercontent.com/3629489/49662163-1b109c80-fa19-11e8-82dd-b6347dae2e1c.png)
![image](https://user-images.githubusercontent.com/3629489/49662210-2fed3000-fa19-11e8-9e26-cb2288eae22f.png)

**Comment by ndrscrls** [comment]
yes, that's exactly what I'm doing. The problem is with git bash, I tried changing the `Shell to run` setting to cmd, and it does work, but when I switch back to bash the `cd` command is never invoked.

**Comment by vbjay** [comment]
You did close GE and reload it after changing the shell option, right?

**Comment by ndrscrls** [comment]
yes.

**Comment by vbjay** [comment]
Switched to cmd and closed/opened and switched repo

![image](https://user-images.githubusercontent.com/3629489/49662626-5bbce580-fa1a-11e8-91c8-93e10d514f32.png)

Switched it back to bash and redid process

![image](https://user-images.githubusercontent.com/3629489/49662756-b35b5100-fa1a-11e8-9929-00d00635d96c.png)

**Comment by vbjay** [comment]
Close GE.  Use task manager to look for any conemu*.exe or git.exe or bash.exe processes.  Kill all.  Run GE again and try again.  
![image](https://user-images.githubusercontent.com/3629489/49663180-d5090800-fa1b-11e8-8fc0-7a98eba61fbb.png)

**Comment by ndrscrls** [comment]
I tried it, I didn't work, I get the same behavior.

**Comment by vbjay** [comment]
Installed or portable version?

**Comment by ndrscrls** [comment]
Installed version.

**Comment by vbjay** [comment]
Open settings to this page and show us that all is green.  

![image](https://user-images.githubusercontent.com/3629489/49664095-da1b8680-fa1e-11e8-9b84-060eaebe0509.png)

**Comment by ndrscrls** [comment]
![image](https://user-images.githubusercontent.com/10979446/49664380-bdcc1980-fa1f-11e8-92ef-4e13fbd61cfc.png)

**Comment by vbjay** [comment]
Download PDB zip from releases on github.  Extract to installed location.  Clone our repo (update all submodules).  Open project in vs.  Open your GE and attach to GE process.  Once debugging, Set breakpoint to here https://github.com/gitextensions/gitextensions/blob/4138d9580c640377ff512ff769c8023a969ccdb1/GitUI/CommandsDialogs/FormBrowse.cs#L2868  Step through and find out if it hits teh return.

**Comment by ndrscrls** [comment]
All the repos are located in a C:\Projects folder, they are out of the HOME(~) folder.

**Comment by ndrscrls** [comment]
Ok so I downloaded the GitExtensions code and debugged the `ChangeTerminalActiveFolder` method inside FormBrowse.cs.  `path` is not `null` and terminal `_terminal?.RunningSession` is also not `null`, so it is executing `ClearTerminalCommandLineAndRunCommand("cd " + posixPath);` `posixPath` is also correct. I stepped in all the way to GuiMacroExecutor.cs:ExecuteInProcessAsync in the ConEmuWinForms project, I'm getting exit code CERR_GUIMACRO_SUCCEEDED(133) when the `Paste` macro is run in the console session, the expected result is 0 but this is not considered an error in the code, still, nothing happens in the UI. So this is a ConEmuWinForms problem, and it might be specific to my machine, there is nothing to do on the GitExtensions side. Thanks for all your help and have a good day.
